### PR TITLE
Update Metrics endpoints for ODH operator

### PR DESCRIPTION
**Pull request by anishasthana**
Signed-off-by: Anish Asthana <anishasthana1@gmail.com>

### CODE REVIEW

**Review by 4n4nd — APPROVED** [review]

File: prometheus/operator/base/openshift-operators-rbac/prometheus-monitoring-role.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    resources:
+      - services
+      - endpoints
+      - pods
+  - verbs:
+      - get
+    apiGroups:
+      - ''
+    resources:
+      - configmaps
```
Comment (crobby) [inline prometheus/operator/base/openshift-operators-rbac/prometheus-monitoring-role.yaml:23]: newline

**Comment by anishasthana** [comment]
This is now ready for review.

**Comment by anishasthana** [comment]
/retest

I'm assuming this was a flake.
```
level=info msg=Waiting up to 40m0s for the cluster at <url> to initialize...
level=error msg=Cluster operator authentication Degraded is True with IngressStateEndpoints_MissingSubsets::OAuthRouteCheckEndpointAccessibleController_SyncError::OAuthServerDeployment_DeploymentAvailableReplicasCheckFailed::OAuthServerRoute_InvalidCanonicalHost::OAuthServiceCheckEndpointAccessibleController_SyncError::OAuthServiceEndpointsCheckEndpointAccessibleController_SyncError::OAuthVersionDeployment_GetFailed::Route_InvalidCanonicalHost::WellKnownReadyController_SyncError: IngressStateEndpointsDegraded: No subsets found for the endpoints of oauth-server
```

**Comment by anishasthana** [comment]
ODH metrics can be seen be running `{job="opendatahub-operator-metrics"}` as a prometheus query

**Pull request merged**